src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -54,21 +72,34 @@ boot_comp_pgtbl_expand(size_t n_pte, pgtblcap_t pt, vaddr_t vaddr, struct cobj_h
 	}
 
 	for (i = 0; i < n_pte; i++) {
-		if (!cos_pgtbl_intern_alloc(&boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
+		if (!cos_pgtbl_intern_alloc(boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
 	}
 }
 
+#define RESMGR_UNTYPED_MEM_SZ (COS_MEM_KERN_PA_SZ - (4 * PGD_RANGE))
+
 /* Initialize just the captblcap and pgtblcap, due to hack for upcall_fn addr */
 static void
 boot_compinfo_init(spdid_t spdid, captblcap_t *ct, pgtblcap_t *pt, u32_t vaddr)
 {
-	*ct = cos_captbl_alloc(&boot_info);
+	*ct = cos_captbl_alloc(boot_info);
 	assert(*ct);
-	*pt = cos_pgtbl_alloc(&boot_info);
+	*pt = cos_pgtbl_alloc(boot_info);
 	assert(*pt);
 
-	new_comp_cap_info[spdid].compinfo = &new_compinfo[spdid];
-	cos_compinfo_init(new_comp_cap_info[spdid].compinfo, *pt, *ct, 0, (vaddr_t)vaddr, 4, &boot_info);
+	/* FIXME: some of the data-structures are redundant! can be removed! */
+	new_comp_cap_info[spdid].defci = &new_comp_cap_info[spdid].def_cinfo;
```

Comment:
So you got rid of `new_compinfo` and replaced with the compinfo inlined into the `new_comp_cap_info`.  Can you comment on why you made this change?

Reply:
`new_comp_cap_info[spdid].compinfo = &new_compinfo[spdid]` => This is really unnecessary! We have 2 structs (both preallocated) and we use same index in both structs and initialize a pointer in `new_comp_cap_info` to point to that `new_compinfo`. 
Cleaner way is to have `cos_compinfo` struct inlined into the outer struct imo.